vault/audited_headers.go

```diff
@@ -0,0 +1,116 @@
+package vault
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+const (
+	// Key used in the BarrierView to store and retrieve the header config
+	auditedHeadersEntry = "audited-headers"
+	// Path used to create a sub view off of BarrierView
+	auditedHeadersSubPath = "audited-headers-config/"
+)
+
+type auditedHeaderSettings struct {
+	HMAC bool `json:"hmac"`
+}
+
+// AuditedHeadersConfig is used by the Audit Broker to write only approved
+// headers to the audit logs. It uses a BarrierView to persist the settings.
+type AuditedHeadersConfig struct {
+	Headers map[string]*auditedHeaderSettings
+
+	view *BarrierView
+	sync.RWMutex
+}
+
+// add adds or overwrites a header in the config and updates the barrier view
+func (a *AuditedHeadersConfig) add(header string, hmac bool) error {
```

Comment:
[Optional] Since there is only one setting and since this is an internal implementation, I guess this looks okay. I was wondering if this should take in `*auditedHeaderSettings`.